### ISSUE TITLE
The results are bad？

### ISSUE THREAD

**Issue by YoungSeng** [GestureGeneration/Speech_driven_gesture_generation_with_autoencoder#15]
Thank you for your wonderful work.

 I followed the steps step by step to reproduce.

`...`
`205600/206899 [============================>.] - ETA: 2s - loss: 0.0013`
`
206899/206899 [==============================] - 422s 2ms/step - loss: 0.0013 - val_loss: 8.2411e-04
`

![model_500ep_posvel_60](https://user-images.githubusercontent.com/37477030/169937284-0cf26812-aba9-4d5c-b6ce-422f7713af7d.png)

But it's not ideal to use the visualization tools you provide:

![1094](https://user-images.githubusercontent.com/37477030/169938363-589872cf-4550-4009-81c1-ec4e50d44f6e.png)

Can you help me see what the problem is? Thanks!

**Comment by Svito-zar** [comment]
Dear @YoungSeng  if you are currently working on the GENEA Challenge 2022- you should use this fork of the repo:
https://github.com/genea-workshop/Speech_driven_gesture_generation_with_autoencoder/tree/GENEA_2022

And here is the page about visualization tools for the challenge data:
https://github.com/genea-workshop/genea_challenge_2022/tree/main/visualizations

Hope that helps

**Comment by YoungSeng** [comment]
Yes, but I haven't got link for teh dataset yet, I simply want to reproduce your previous work first, is the visualization tool incorrect for the previous work? Or I get something wrong?

Thanks a lot

**Comment by Svito-zar** [comment]
that visualization tool only works for the Japanese dataset we used for that paper. are you using that dataset?

**Comment by YoungSeng** [comment]
Yes, I use the same dataset and all steps are the same, and /data I get is:
![image](https://user-images.githubusercontent.com/37477030/169992129-ebeba2b6-040c-4396-b877-d0c671e44521.png)

**Comment by Svito-zar** [comment]
@YoungSeng, could you please share the motion file that you are trying to visualize?
@DaiHasegawa, could you please check this issue?

**Comment by YoungSeng** [comment]
Thank you very much!

Can you help first to see if the loss is normal? I'll try to upload the file.

Stage1:

<code>Done training for 15 epochs</code>
<code>
The model was saved in file: chkpt/chkpt-final</code>
<code>
|  16 (80.0 %) | 1.29|</code>
<code>
|  17 (85.0 %) | 1.65|</code>
<code>
|  18 (90.0 %) | 1.35|</code>
<code>
|  19 (95.0 %) | 1.48|</code>
<code>
|  20 (100. %) | 1.40|</code>
<code>
Done training for 20 epochs</code>
<code>
The model was saved in file: chkpt/chkpt-final</code>
<code>
|  21 (105. %) | 1.34|</code>
<code>
Done training for 20 epochs, 37716 steps.</code>
<code>
The final model was saved in file: chkpt/chkpt-final</code>
<code>
The training was running for 7.133  min</code>

Stage2:

`205600/206899 [============================>.] - ETA: 2s - loss: 0.0013`
`206899/206899 [==============================] - 422s 2ms/step - loss: 0.0013 - val_loss: 8.2411e-04`

**Comment by Svito-zar** [comment]
Something is weird here, I think. Because the loss of the representation learning (Stage 1) is usually getting much lower and in your case, it does not seem to learn much ... what is the dimensionality of the encoding that you are using?

**Comment by Svito-zar** [comment]
For example, when I train the AutoEncoder on the GENEA 2022 Challenge data the loss is dropping from 32 to 0.3

**Comment by YoungSeng** [comment]
Yes, there are only 20 epochs in the first stage, and the training is done in one minute in the first stage, should I increase the number of stages?

The complete results of the first stage:

```python
Finetune the whole network on  1796  batches with  128  training examples in each for 20  epochs...

 ______________ ______
|     Epoch    | RMSE |
|------------  |------|
|   0 (0.00 %) |351.23|
|   1 (5.00 %) | 4.90|
|   2 (10.0 %) | 1.84|
|   3 (15.0 %) | 1.60|
|   4 (20.0 %) | 1.52|
|   5 (25.0 %) | 1.70|
Done training for 5 epochs
The model was saved in file: chkpt/chkpt-final
|   6 (30.0 %) | 1.43|
|   7 (35.0 %) | 1.59|
|   8 (40.0 %) | 1.61|
|   9 (45.0 %) | 1.41|
|  10 (50.0 %) | 1.52|
Done training for 10 epochs
The model was saved in file: chkpt/chkpt-final
|  11 (55.0 %) | 1.62|
|  12 (60.0 %) | 1.64|
|  13 (65.0 %) | 1.72|
|  14 (70.0 %) | 1.44|
|  15 (75.0 %) | 1.77|
Done training for 15 epochs
The model was saved in file: chkpt/chkpt-final
|  16 (80.0 %) | 1.29|
|  17 (85.0 %) | 1.65|
|  18 (90.0 %) | 1.35|
|  19 (95.0 %) | 1.48|
|  20 (100. %) | 1.40|
Done training for 20 epochs
The model was saved in file: chkpt/chkpt-final
|  21 (105. %) | 1.34|
Done training for 20 epochs, 37716 steps.
The final model was saved in file: chkpt/chkpt-final
The training was running for 7.133  min
```
Second attempt：

```python
Finetune the whole network on  1796  batches with  128  training examples in each for 20  epochs...

 ______________ ______
|     Epoch    | RMSE |
|------------  |------|
|   0 (0.00 %) |351.23|
|   1 (5.00 %) | 4.46|
|   2 (10.0 %) | 1.83|
|   3 (15.0 %) | 1.64|
|   4 (20.0 %) | 1.62|
|   5 (25.0 %) | 1.57|
Done training for 5 epochs
The model was saved in file: chkpt/chkpt-final
|   6 (30.0 %) | 1.40|
|   7 (35.0 %) | 1.47|
|   8 (40.0 %) | 1.55|
|   9 (45.0 %) | 1.50|
|  10 (50.0 %) | 1.57|
Done training for 10 epochs
The model was saved in file: chkpt/chkpt-final
|  11 (55.0 %) | 1.62|
|  12 (60.0 %) | 1.55|
|  13 (65.0 %) | 1.50|
|  14 (70.0 %) | 1.63|
|  15 (75.0 %) | 1.63|
Done training for 15 epochs
The model was saved in file: chkpt/chkpt-final
|  16 (80.0 %) | 1.26|
|  17 (85.0 %) | 1.53|
|  18 (90.0 %) | 1.40|
|  19 (95.0 %) | 1.46|
|  20 (100. %) | 1.41|
Done training for 20 epochs
The model was saved in file: chkpt/chkpt-final
|  21 (105. %) | 1.53|
Done training for 20 epochs, 37716 steps.
The final model was saved in file: chkpt/chkpt-final
The training was running for 6.658  min
```


Thanks for you reply!

**Comment by Svito-zar** [comment]
No need in that. It seems to be training properly

**Comment by Svito-zar** [comment]
could you please share the output from the model that you get?

**Comment by YoungSeng** [comment]
Thanks!

Sorry, I mean should I increase the number of epochs?

Of course I can share the output, I'll try to upload it tomorrow.

**Comment by Svito-zar** [comment]
I don't think that you need to increase the number of epochs, it seems that the training has converged

**Comment by Svito-zar** [comment]
I suspect that maybe you are forgetting to decode the motion after it was generated.
Are you doing this step `python motion_repr_learning/ae/decode.py` ?

**Comment by YoungSeng** [comment]
Hello, there are the results I got.

I did the decode step.

 Can you help me see if there is a problem with model training or visualization?

[output_1094.txt](https://github.com/GestureGeneration/Speech_driven_gesture_generation_with_autoencoder/files/8780487/output_1094.txt)
[output_1168.txt](https://github.com/GestureGeneration/Speech_driven_gesture_generation_with_autoencoder/files/8780484/output_1168.txt)
[predict_1094_20fps.txt](https://github.com/GestureGeneration/Speech_driven_gesture_generation_with_autoencoder/files/8780485/predict_1094_20fps.txt)
[predict_1168_20fps.txt](https://github.com/GestureGeneration/Speech_driven_gesture_generation_with_autoencoder/files/8780486/predict_1168_20fps.txt)

**Comment by YoungSeng** [comment]
Hello, can you take a look for me? Thanks！

**Comment by Svito-zar** [comment]
It is not obvious to me what went wrong for you. I am going to try to reproduce the results again myself and come back to you later

**Comment by Svito-zar** [comment]
@YoungSeng , I see now that your output has 384 dimensions. that means that it still contains velocities, but not only coordinates.
Please apply `python helpers/remove_velocity.py` to it, as indicated at the end of README

**Comment by Svito-zar** [comment]
If I remove the velocities from the file you provided - I get nice results :) 
![Screenshot from 2022-06-03 12-00-15](https://user-images.githubusercontent.com/15908492/171833403-b8c1c73e-0944-4cca-b672-dcf81bcfa5c1.png)

**Comment by Svito-zar** [comment]
Hence your issue was just that you forgot to apply `python helpers/remove_velocity.py`
@YoungSeng, please close this issue if your question was answered

**Comment by YoungSeng** [comment]
Thanks! I forgot this!

Problem solved!
